pyCGM_Single/tests/test_pycgmIO.py

```diff
@@ -0,0 +1,870 @@
+            'RightFootLength': 175.794006347656,'RightHandLength': 87.4593048095703,
+            'RightHandThickness': 17.0,'RightHumerusLength': 290.563262939453,
+            'RightKneeWidth': 120.0,'RightLegLength': 1000.0,
+            'RightRadiusLength': 264.853607177734,'RightShankRotation': 0.0,
+            'RightShoulderOffset': 40.0,'RightSoleDelta': 0.0,
+            'RightStaticPlantFlex': 0.17637075483799,'RightStaticRotOff': 0.03440235927701,
+            'RightThighRotation': 0.0,'RightTibiaLength': 443.718109130859,
+            'RightTibialTorsion': 0.0,'RightWristWidth': 60.0,
+            'STRNz': -207.033920288086,'T10x': -205.628646850586,
+            'T10y': -7.51900339126587,'T10z': -261.275146484375,
+            'Theta': 0.500000178813934,'ThorOx': 1.28787481784821,
+            'ThorOy': 0.0719171389937401,'ThorOz': 499.705780029297
+        }
+
+        #Test that loadVSK correctly returned as a dictionary
+        assert isinstance(result_vsk, dict)
+        #Test accurate loading
+        np.testing.assert_equal(result_vsk, expected_result)
+
+    def test_loadVSK_exceptions(self):
+        """
+        Test that loading a non-existent file raises an
+        exception.
+        """
+        with pytest.raises(Exception):
+            pycgmIO.loadVSK("NonExistentFilename")
+
+    @pytest.mark.parametrize("motionData, expected_labels, expected_values", [
+        (
+         [{'A': [1, 2, 3], 'B': [4, 5, 6]},
+          {'A': [2, 3, 4], 'B': [5, 6, 7]}],
+         ['A', 'B'],
+         [np.array([[1, 2, 3],[4, 5, 6]]), np.array([[2, 3, 4],[5, 6, 7]])]
+        ),
+        (
+         [{'A': np.array([1, 2, 3]), 'B': np.array([4, 5, 6])},
+          {'A': np.array([2, 3, 4]), 'B': np.array([5, 6, 7])}],
+         ['A', 'B'],
+         [np.array([[1, 2, 3],[4, 5, 6]]), np.array([[2, 3, 4],[5, 6, 7]])]
+        ),
+        (
+         [{'A': np.array([1, 2, 3]), 'B': np.array([4, 5, 6])},
+          {'A': np.array([2, 3, 4])}],
+         ['A', 'B'],
+         [np.array([[1, 2, 3],[4, 5, 6]]), np.array([[2, 3, 4]])]
+        ),
+        (
+         [{'B': np.array([4, 5, 6])},
+          {'A': np.array([2, 3, 4])}],
+         ['B'],
+         [np.array([[4, 5, 6]]), np.array([[2, 3, 4]])]
+        )
+    ])
+    def test_splitDataDict_accuracy(self, motionData, expected_labels, expected_values):
+        """
+        This function tests pycgmIO.splitDataDict(motionData),
+        where motionData is a list of dictionaries of motion capture data.
+        This function splits the motionData into a tuple of values, labels.
+
+        We tests cases where values are lists or numpy arrays.
+        We demonstrate unexpected behavior that the function produces when
+        keys are not present in every dictionary of motiondata.
+        """
+        result_values, result_labels = pycgmIO.splitDataDict(motionData)
+        assert result_labels == expected_labels
```

Comment:
what is the difference between these two lines?

Reply:
Line 799 is testing for equality of the returned labels, and line 800 is testing for equality between the returned values. I split it up because splitDataDict returns a tuple of values, labels.